site/content/en/docs/Getting started/first_kustomization.md

```diff
@@ -57,10 +107,251 @@ spec:
         app: nginx
     spec:
       containers:
-      - name: nginx
-        image: nginx:latest
+      - image: nginx
+        name: nginx
         ports:
         - containerPort: 80
+```
+
+### What are Bases and Overlays?
+
+Let's think of a scenario where we need to deploy the nginx manifests from the previous section to 2 environments called `Staging` and `Production`, however although most of the manifest is the same for both environments we need to apply some environment specific changes.  Traditionally we could duplicate the manifests and apply the changes manually or rely on some templating engine. With kustomize we can avoid templating and duplication of our manifests and apply the different changes we need using overlays.  With this approach, the `base` would contain the common part of the our manifests and the `overlays` contain our environment specific changes.
+
+To continue with this tutorial, create the `kustomization.yaml` files for our two overlays:
+
+```bash
+cat <<'EOF' >overlays/staging/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../base
 EOF
+
+cat <<'EOF' >overlays/production/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../base
+EOF
+```
+
+The `kustomization.yaml` files in the overlays are just including the `base` folder, so if you were to run `kustomize build` on the overlay folders at this point you would get the same output we got when we built `base`. The directory structure you created so far should look like this:
+
+```
+kustomize-example
+├── base
+│   ├── deployment.yaml
+│   ├── kustomization.yaml
+│   └── service.yaml
+└── overlays
+    ├── production
+    │   └── kustomization.yaml
+    └── staging
+        └── kustomization.yaml
+```
+
+### Customising our overlays
+
+For the purposes our example, let's define some requirements of how our deployment should look like in the two environments:
+
+|Requirement|         Production         |         Staging         |
+|-----------|----------------------------|-------------------------|
+|Name       |overlay1-nginx-production   |overlay2-nginx-staging   |
+|Namespace  |production                  |staging                  |
+|Replicas   |3                           |2                        |
+
+
+We can achieve the names required by making use of `namePrefix` and `nameSuffix` as follows:
```

Comment:
Do we have an actual name for things like `namePrefix`, `nameSuffix`, `namespace`, etc as we define in a `kustomization.yaml` file? Directives, commands, something else?

Reply:
I avoided using a name to refer to these because I was unsure if they had one... @natasha41575 ?

Reply:
> Do we have an actual name for things like namePrefix, nameSuffix, namespace, etc as we define in a kustomization.yaml file? Directives, commands, something else?

I usually call them "fields in the kustomization file", "kustomization fields", etc

Reply:
I saw in some of the docs that they are referred to as fields in some cases and I like the sound of directives for these more than fields, but thinking more about it maybe we have to be a bit careful about introducing new terms since they can also be referred to as transformers? Or define that directives and transformers may be used interchangeably?

Reply:
My two cents: They can be "fields" when you're talking about the literal key you put into a Kustomization file, "transformers" when you're talking about the code that drives their functionality, or maybe "features" when you're being really abstract. "Directives" is already a (somewhat obscure) feature in this space: [patch directives](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md#basic-patch-format).